src/Storages/Statistics/StatisticsMinMax.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Storages/Statistics/StatisticsMinMax.h>
+#include <Common/FieldVisitorConvertToNumber.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Interpreters/convertFieldToType.h>
+
+#include <algorithm>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+}
+
+StatisticsMinMax::StatisticsMinMax(const SingleStatisticsDescription & statistics_description, const DataTypePtr & data_type_)
+    : IStatistics(statistics_description)
+    , data_type(data_type_)
+{
+}
+
+void StatisticsMinMax::update(const ColumnPtr & column)
+{
+    for (size_t row = 0; row < column->size(); ++row)
+    {
+        if (column->isNullAt(row))
+            continue;
+
+        auto value = column->getFloat64(row);
+        min = std::min(value, min);
+        max = std::max(value, max);
+    }
+    row_count += column->size();
+}
+
+void StatisticsMinMax::serialize(WriteBuffer & buf)
+{
+    writeIntBinary(row_count, buf);
+    writeFloatBinary(min, buf);
+    writeFloatBinary(max, buf);
+}
+
+void StatisticsMinMax::deserialize(ReadBuffer & buf)
+{
+    readIntBinary(row_count, buf);
+    readFloatBinary(min, buf);
+    readFloatBinary(max, buf);
+}
+
+Float64 StatisticsMinMax::estimateLess(const Field & val) const
+{
+    Field val_converted = convertFieldToType(val, *data_type);
```

Comment:
I believe this is where https://github.com/ClickHouse/ClickHouse/issues/67742 hits.

For example, if you run HanFei's [test case](https://github.com/ClickHouse/ClickHouse/pull/67858/files#diff-383c270da8abeb37e28e49f3f80adba21ca39a837c60af1840655a95a6bb4b8e), is it correct that `*data_type` is `DataTypeFloat64` (the type of column `a`) and that `val_converted` will have `isNull` == 0 (because we can't convert `val` which is a `String` to a `Float64`)? In this case, we would return 0 in l. 58. (kindly check). Note that returning 0 is incorrect in this situation, we really need to return a status "cannot make an estimation". So we'd need to return (for example) `std::optional<Float64>`.

Reply:
If `*data_type` is `DataTypeFloat64` `val_converted ` will be 10 with type is ` Float64`. `convertFieldToType ` will do implicity type conversion, for example: '2024-01-01' can be converted to a `Date` value, '1' can be converted to a `(U)Int*` or `Float*` value.